lib/Doctrine/ORM/AbstractQuery.php

```diff
@@ -381,6 +381,14 @@ public function setParameters($parameters)
      */
     public function setParameter($key, $value, $type = null)
     {
+        if ($type === null) {
```

Comment:
> We should assume every object passed must be an entity or specify a $type as third argument. Only primitive types should work with type inference.

I think the deprecation should fire only in some cases, but I'm not sure if this is the right way to do it: can we pass objects that are not entities and not rely on type inference here @beberlei ?

```suggestion
        if (!is_object($value) && $type === null) {
```

Reply:
Based on the example of `Datetime` I guess we should also check if the value is an object and type is null, than we trigger the deprecation, but let's wait and see if we can get some more input